# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1661619303 -7200
# Parent  dcfb34cf898f15f69bc98e6b4df2b93924738ac4
Feature: protect against failure of nsIMsgFolder.msgDatabase.

diff --git a/mail/modules/MailUtils.jsm b/mail/modules/MailUtils.jsm
--- a/mail/modules/MailUtils.jsm
+++ b/mail/modules/MailUtils.jsm
@@ -614,19 +614,23 @@ var MailUtils = {
    * @param {nsIMsgFolder} folder - The folder to check.
    * @returns {nsIMsgDBHdr}
    */
   findMsgIdInFolder(msgId, folder) {
     let msgHdr;
 
     // Search in folder.
     if (!folder.isServer) {
-      msgHdr = folder.msgDatabase.getMsgHdrForMessageID(msgId);
-      if (msgHdr) {
-        return msgHdr;
+      try {
+        msgHdr = folder.msgDatabase.getMsgHdrForMessageID(msgId);
+        if (msgHdr) {
+          return msgHdr;
+        }
+      } catch (ex) {
+        Cu.reportError(`Database for ${folder.name} not accessible`);
       }
     }
 
     // Search subfolders recursively.
     for (let currentFolder of folder.subFolders) {
       msgHdr = this.findMsgIdInFolder(msgId, currentFolder);
       if (msgHdr) {
         return msgHdr;
